packages/playwright-file-snapshots/tests/repeatable-snapshot.spec.ts

```diff
@@ -0,0 +1,142 @@
+    const currentTime = performance.now();
+    this.snapshotIntervals.push(currentTime - this.intervalStartTime);
+    this.intervalStartTime = currentTime;
+  }
+}
+
+function assertSnapshotIntervals(
+  expectedIntervals: number[],
+  instrumentation: SnapshotInstrumentation,
+): void {
+  expect(instrumentation.snapshotCount).toBe(expectedIntervals.length);
+
+  for (const [intervalIndex, expectedInterval] of expectedIntervals.entries()) {
+    const actualInterval = instrumentation.snapshotIntervals.at(intervalIndex)!;
+    expect(actualInterval).toBeGreaterThanOrEqual(expectedInterval);
+    expect(actualInterval).toBeLessThan(expectedInterval + 50);
+  }
+}
+
+test("when validation file is missing, waits for delay before creating snapshot", async () => {
+  cleanTmpDir();
+
+  const testExpect = defineValidationFileExpect({
+    validationDir: TMP_VALIDATION_DIR,
+    outputDir: TMP_OUTPUT_DIR,
+    soft: false,
+  });
+
+  const instrumentation = new SnapshotInstrumentation();
+  await expect(() =>
+    testExpect(() => {
+      instrumentation.addSnapshot();
+      return "missing value";
+    }).toMatchJsonFile(),
+  ).rejects.toThrowError();
+
+  assertSnapshotIntervals([250], instrumentation);
+});
+
+test(
+  "when snapshot updates are enabled, waits for delay before creating snapshot",
+  { tag: ["@snapshot-update"] },
+  async () => {
+    test.skip(
+      test.info().config.updateSnapshots !== "changed",
+      "Snapshot updates are disabled",
+    );
+
+    const testExpect = defineValidationFileExpect({
+      soft: false,
+    });
+
+    const instrumentation = new SnapshotInstrumentation();
+    await expect(() =>
+      testExpect(() => {
+        instrumentation.addSnapshot();
+        return "changed value";
+      }).toMatchJsonFile(),
+    ).rejects.toThrowError();
+
+    assertSnapshotIntervals([250], instrumentation);
+  },
+);
+
+test("when validation file exists, repeats failing snapshot until it matches", async () => {
+  const testExpect = defineValidationFileExpect();
+
+  const instrumentation = new SnapshotInstrumentation();
+  await testExpect(() => {
+    if (instrumentation.snapshotCount === 3) {
+      return "stable value";
+    } else {
+      instrumentation.addSnapshot();
+      return `unstable value ${instrumentation.snapshotCount}`;
+    }
+  }).toMatchJsonFile();
+
+  assertSnapshotIntervals([0, 100, 250], instrumentation);
+});
+
+test("when validation file exists, repeats failing snapshot until timeout", async () => {
+  cleanTmpDir();
+
+  const testExpect = defineValidationFileExpect({
+    soft: false,
+  }).configure({ timeout: 500 });
+
+  const instrumentation = new SnapshotInstrumentation();
+  await testExpect(() =>
+    testExpect(() => {
+      instrumentation.addSnapshot();
+      return `unstable value ${instrumentation.snapshotCount}`;
+    }).toMatchJsonFile(),
+  ).rejects.toThrowError();
+
+  assertSnapshotIntervals([0, 100, 250, 500], instrumentation);
```

Comment:
You could add a test to test the max delay of 1000 between retries

Reply:
My initial intention was to keep the execution time low. But for now it should not matter too much. I extended the delay to cover two more intervals with 1s.